compiler/circle-resizer/include/ModelData.h

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2025 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __CIRCLE_RESIZER_MODEL_DATA_H__
+#define __CIRCLE_RESIZER_MODEL_DATA_H__
+
+#include "Shape.h"
+
+#include <string>
+#include <memory>
+#include <vector>
+
+namespace luci
+{
+class Module;
+}
+
+namespace circle_resizer
+{
+
+/**
+ * The representation of Circle Model.
+ * The purpose of the class is to keep the buffer and the module representation of the model
+ * synchronized.
+ */
+class ModelData
+{
+public:
+  /**
+   * @brief Initialize the model with buffer representation.
+   *
+   * Exceptions:
+   * - std::runtime_error if interpretation of provided buffer as a circle model failed.
+   */
+  explicit ModelData(const std::vector<uint8_t> &buffer);
+
+  /**
+   * @brief Initialize the model with buffer representation.
+   *
+   * Exceptions:
+   * - std::runtime_error if reading a model from provided path failed.
+   */
+  explicit ModelData(const std::string &model_path);
+
+  /**
+   * @brief Dtor of ModelData. Note that explicit declaration is needed to satisfy forward
+   * declaration + unique_ptr.
+   */
+  ~ModelData();
+
+  /**
+   * @brief Notify that the buffer representation of the model has been modified so the module is no
+   * more valid.
+   */
+  void invalidate_module();
+
+  /**
+   * @brief Notify that the module representation of the model has been modified so the buffer is no
+   * more valid.
+   */
+  void invalidate_buffer();
```

Comment:
Buffer representation of the model (after `circle::GetMutableModel`) is need to change the inputs shapes like shown in https://github.com/Samsung/ONE/pull/14727/files#diff-89115d8f21eafbf362d2bfbe1ea5677c609681b09b0ad094d6eec8a8e6e5a27eR44